tests/xtro-sharpie/Helpers.cs

```diff
@@ -209,6 +209,62 @@ public static bool IsAvailable (this Decl decl)
 			return result;
 		}
 
+		public static bool IsDeprecated (this Decl decl)
+		{
+			// there's no doubt we need to ask for the current platform
+			var result = decl.IsDeprecated (Platform);
+
+			// some categories are not decorated (as not available) but they extend types that are
+			if (!result.HasValue) {
+				// first check if we're checking the category itself
+				var category = decl as ObjCCategoryDecl;
+				if (category != null)
+					result = category.ClassInterface.IsDeprecated (Platform);
+
+				if (!result.HasValue) {
+					// then check if we're a method inside a category
+					category = (decl.DeclContext as ObjCCategoryDecl);
+					if (category != null)
+						result = category.ClassInterface.IsDeprecated (Platform);
+				}
+			}
+
+			// but right now most frameworks consider tvOS, watchOS, and catalyst like iOS unless 
+			// decorated otherwise so we must check again if we do not get a definitve answer
+			if ((result == null) && ((Platform == Platforms.tvOS) || (Platform == Platforms.watchOS) || (Platform == Platforms.MacCatalyst)))
+				result = decl.IsDeprecated (Platforms.iOS);
```

Comment:
I don't understand why this logic was not moved to the other IsDeprecated method and then simply not return a bool? form there.

Reply:
This method is basically a copy of the `IsAvailable` methods above - but in any case, moving this logic inside the other IsDeprecated method and returning a plain `bool` would break the logic just above here wrt categories